### PR TITLE
Edit Time class to allow more time input formats

### LINKED ISSUE #101: Update events to accept more time formats

### LINKED ISSUE #202: [PE-D] [Athena] Event date editing is tedious
As of now users have to type the whole `dd-MM-yyyy HH:mm` format, perhaps it would allow them to be faster if you allowed for other formats such as `d-M-yy HH:mm`.

Or better yet, allow `d-M-yy` inputs or `HH:mm` inputs:
The former keeps the time of the event while changing the date, the latter keeps the date and changes the time.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: StopTakingAllTheNames/ped#4

### PR DESCRIPTION
**Pull request by solkim-83**
Previously, only the following format for event time was allowed as an user input:
1. dd-MM-yyyy HH:mm 

I've changed the Time class to accept more formats such as:
1. dd-MM-yyyy HH:mm
2. yyyy-MM-dd HH:mm
3. and such that [.][/][-] can be used interchangeably to key in the date and time.

### CODE REVIEW

File: src/main/java/seedu/address/model/event/Time.java [hunk 1]
```diff
@@ -17,9 +17,14 @@
...
-            + "dd-MM-yyyy HH:mm";
+            + "dd-MM-yyyy HH:mm\n"
+            + "yyyy-dd-MM HH:mm\n"
+            + "[.][/][-][:] can be used interchangeably to indicate date, time, year of the event.";
 
-    private static final String STANDARD_TIME_PATTERN = "dd-MM-yyyy HH:mm";
+    private static final String symbol = "[.][/][-][:]";
+    private static final String STANDARD_TIME_PATTERN =
+            "[dd" + symbol + "MM" + symbol + "yyyy HH" + symbol + "mm]"
+                    + "[yyyy" + symbol + "dd" + symbol + "MM HH" + symbol + "mm]";
```
Comment (fyshhh) [inline src/main/java/seedu/address/model/event/Time.java:27]: i think this straight up breaks the parser as it relies on this field to check if the input format is correct. i think this branch should only be merged when we actually change `isValidTime` and `parse` below.

Comment (solkim-83) [reply src/main/java/seedu/address/model/event/Time.java:27]: Why do isValidTime and parse need to be changed?

Comment (fyshhh) [reply src/main/java/seedu/address/model/event/Time.java:27]: sorry, this is a misinterpretation on my part.

File: src/main/java/seedu/address/logic/commands/events/AddEventCommand.java [hunk 2]
```diff
@@ -41,7 +41,7 @@
             + "[" + PREFIX_ADD_PERSON + "CONTACT_INDEX_LIST]\n"
             + "Example: " + COMMAND_WORD + " " + COMMAND_TYPE + " "
             + PREFIX_DESCRIPTION + "CS2103 Team meeting" + " "
-            + PREFIX_DATETIME + "12-12-1234 12:34"
+            + PREFIX_DATETIME + "12-10-20 12:00 "
```
Comment (fyshhh) [inline src/main/java/seedu/address/logic/commands/events/AddEventCommand.java]: i think this is inconsistent with the actual pattern below.

**Review by LinkedInk — APPROVED** [review]
lgtm

**Pull request merged**